packages/driver/src/cross-origin/cypress.ts

```diff
@@ -148,7 +181,13 @@ const onBeforeAppWindowLoad = (Cypress: Cypress.Cypress, cy: $Cy) => (autWindow:
     onSubmit (e) {
       return Cypress.action('app:form:submitted', e)
     },
-    onBeforeUnload (e) {
+    async onBeforeUnload (e) {
+      // The before unload event is propagated to primary through code injected into the AUT.
+      // We need to sync this state value prior to changing stability otherwise we will erroneously log a loading event.
+      const duringUserTestExecution = await Cypress.specBridgeCommunicator.toPrimaryPromise('sync:during:user:test:execution')
+
+      cy.state('duringUserTestExecution', duringUserTestExecution)
+
```

Comment:
onBeforeLoad could happen at any time during a command queue execution, post messaging with the primary cypress instance would delay setting stabilty which may allow the command queue to progress further than without the delay, however cy.isStable is the command that kicks off the need to have an up to date state value.

It might be possible to allow stability to be set and update 'stabilityChanged' in the navigation command to await getting this value if it's a spec bridge.

https://github.com/cypress-io/cypress/blob/matth/cy-origin-async-attach/packages/driver/src/cy/commands/navigation.ts#L278

If thats preferable i can look into that after i've addressed all other comments.

Reply:
I went ahead and did this, it seems like a better option.

Reply:
Can you link the code you changed to address this? Still seems like there's a race condition if we're delaying setting stability to false.

Reply:
Here is where the code moved to.
https://github.com/cypress-io/cypress/blob/777d52449c3763ec255cb7b7eabf1e1494ac8a2c/packages/driver/src/cy/commands/navigation.ts#L276-L281

Reply:
Since we now await after setting stability to false we should not be exacerbating the inherent race condition when setting stability to false.